jpa/subsystem/src/main/java/org/jboss/as/jpa/config/PersistenceUnitMetadataImpl.java

```diff
@@ -381,11 +383,14 @@ public List<ClassTransformer> getTransformers() {
 
     @Override
     public void addTransformer(ClassTransformer classTransformer) {
-        transformers.add(classTransformer);
-        if (ROOT_LOGGER.isTraceEnabled()) {
-            ROOT_LOGGER.tracef("added entity class transformer '%s' for '%s'",
-                    classTransformer.getClass().getName(),
-                    getScopedPersistenceUnitName());
+        // WFLY-20393 Ensure that only one bytecode transformer is added for each Persistence Unit
+        if (true == coordinateDelegatingTransformerAdd.compareAndSet(false, true)) {
```

Comment:
Not a big deal, but the `true ==` is redundant.

Reply:
fixed